HT7_1.py

```diff
@@ -0,0 +1,296 @@
+
+    def __save_words_cnt_to_csv(self):
+        with open('cnt_words.csv', 'w') as csvfile:
+            for key in self.words_dict.keys():
+                csvfile.write("%s- %s\n" % (key, self.words_dict[key]))             #delimiter is "-" as it was requested in russian version of lesson
+
+    def __count_letters_from_final_file(self):
+        with open(f"{self.file_to_count}") as file:
+            letter_list = [word for line in file for word in line if word.isalpha()]
+        l_dict = {i.lower(): [letter_list.count(i.upper()) + letter_list.count(i.lower()),
+                              letter_list.count(i.upper()),
+                              f"{round(letter_list.count(i.upper()) / (letter_list.count(i.upper()) + letter_list.count(i.lower())), 2)} %"]
+                  for i in sorted(letter_list)}
+        return l_dict
+
+    def __save_letter_cnt_to_csv(self):
+        with open('cnt_letters.csv', 'w', newline='') as csvfile:
+            dw = csv.writer(csvfile, delimiter=',')
+            dw.writerow(["letter", "count_all", "count_uppercase", "percentage"])
+            dw.writerows(
+                [letter, self.letter_dict.get(letter, '0')[0], self.letter_dict.get(letter, '0')[1],
+                 self.letter_dict.get(letter, '0.0 %')[2]]
+                for letter in self.letter_dict.keys())
+
+
+def get_publication_method():
+    while True:
+        publication_method = input("Please select publication method\n1 - type via cli\n2 - via txt file\n")
+        if publication_method == "1":
+            return ReaderCli()
+        elif publication_method == "2":
+            return ReaderTxt()
+        else:
+            print("Wrong format. Please try again or Ctrl+C to exit.")
+
+
+def write_to_list(publication, body):
+    publication_list = []
+    publication_list.insert(0, publication.publication_type)
+    publication_list.insert(1, body)
+    if publication.publication_type == "News":
+        publication_list.insert(2, f'{publication.location}, {publication.get_publish_date()}')
+    elif publication.publication_type == "Ad":
+        publication_list.insert(2, publication.expiration_date)
+    elif publication.publication_type == "Joke":
+        publication_list.insert(2, publication.funny_meter)
+    return publication_list
+
+
+def write_to_file(publication_list):
+    file = open('myfile.txt', 'a+')
+    file.write('---------------------------------' + '\n')
+    file.write(f'{publication_list[0]}\n{publication_list[1]}\n{publication_list[2]}\n')
+    file.write('---------------------------------' + '\n\n')
+    print(f"\nYou publication was added successfully.")
+
+
+counter = Counter("myfile.txt")
+counter.file_size_before = counter.get_file_size()
+
+reader = get_publication_method()
+
+reader.file_name = ""
+
+if reader.publication_method == "txt":
+    reader.file_name = reader.get_file_name()
+    reader.input_list = reader.read_from_txt()
+
+pub = reader.get_publication_type()

```

Comment:
TO REVIEW:
maybe change get_publication_type to smth like `get_publication_template`? Because now it is not clear why we request pub type but getting whole publication instead